### DISCUSSION TITLE
Offline Mode - Spotify and YT Music

### DISCUSSION THREAD

**Discussion by KevinKTM** [music-assistant/support#2652] [Q&A] [answered]
I run a Home Assistant instance on my boat so my internet access is limited to at the dock. Are there any plans to support offline mode fo9r Spotify and/or YT Music (downloading platlists, albums etc) directly my server similar to what I can do on my phone?

**Comment by OzGav** [comment, marked as answer]
No downloading music would violate the T&Cs of those providers.

**Reply by KevinKTM** [reply]
Is downloading supported by the Spotify API? Not suggesting pirating but using their existing authentication system to support an offline mode which I assume requires logging into your account periodically to re-authenticate music downloaded to the device.

Edit:
https://developer.spotify.com/blog/2019-11-27-offline-mobile-sdk

**Reply by OzGav** [reply]
That link is for mobile devices only.
